src/class/objects.php

```diff
@@ -1289,7 +1289,31 @@ function prepareData ($zData = '', $sView = 'list')
                     list($nID, $sName, $sEmail, $sInstitute, $sDepartment, $sCountryID) = $zData['analyzer'];
                     $zData['analysis_by_'] = '<SPAN class="custom_link" onmouseover="lovd_showToolTip(\'' . addslashes('<TABLE border=0 cellpadding=0 cellspacing=0 width=350 class=S11><TR><TH valign=top>User&nbsp;ID</TH><TD>' . ($_AUTH['level'] < LEVEL_MANAGER? $nID : '<A href=users/' . $nID . '>' . $nID . '</A>') . '</TD></TR><TR><TH valign=top>Name</TH><TD>' . $sName . '</TD></TR><TR><TH valign=top>Email&nbsp;address</TH><TD>' . str_replace("\r\n", '<BR>', lovd_hideEmail($sEmail)) . '</TD></TR><TR><TH valign=top>Institute</TH><TD>' . $sInstitute . '</TD></TR><TR><TH valign=top>Department</TH><TD>' . $sDepartment . '</TD></TR><TR><TH valign=top>Country</TH><TD>' . $sCountryID . '</TD></TR></TABLE>') . '\', this);">' . $sName . '</SPAN>';
                 }
-                // In LOVD+, we disable the feature of coloring hidden and marked data, since all data is hidden.
+            }
+            // Status coloring will only be done, when we have authorization.
+            // Instead of having the logic in separate objects and the custom VL object, put it together here.
+            // In LOVD+, we disable the feature of coloring hidden and marked data, since all data is hidden.
+            if (!LOVD_plus && $_AUTH['level'] >= LEVEL_COLLABORATOR) {
+                // Mark row according to the lowest status; Marked is red; lower will be gray.
+                // NOTE: Do we need to worry about adding values here that are defaults?
+                // I'd rather not copy $zData...
+                $zData += array(
+                    'statusid' => STATUS_OK,
+                    'var_statusid' => STATUS_OK,
+                    'ind_statusid' => STATUS_OK,
+                );
+                // The individual's status can be empty, if there is none,
+                //  but that doesn't mean it's non-public.
+                $zData['ind_statusid'] = ($zData['ind_statusid']?: STATUS_OK);
+                // In the VariantOnTranscriptUnique view the var_statusid can contain multiple IDs, these IDs are separated by a ",".
+                // PHP always takes the first integer-like part of a string when a string and an integer are compared.
+                // But to avoid problems in the future, only the first character is compared.
+                $nStatus = min($zData['statusid'], substr($zData['var_statusid'], 0, 1), $zData['ind_statusid']);
+                $zData['class_name'] = '';
+                if ($nStatus <= STATUS_MARKED) {
+                    $zData['class_name'] = ($nStatus == STATUS_MARKED ? 'marked' : 'del');
+                }
+            } else {
```

Comment:
I tried to make this section a bit more concise and the comments more to the point. Here's what I came up with:

~~~php
            // Determine collective status for all data in current viewlist row.
            $nRowStatus = STATUS_OK;
            if (!LOVD_plus && $_AUTH['level'] >= LEVEL_COLLABORATOR) {
                // Loop through possible status fields.
                foreach (array('statusid', 'var_statusid', 'ind_statusid') as $sField) {
                    if (isset($zData[$sField]) && !empty($zData[$sField])) {
                        // Take minimum value of all status fields.
                        // Note: var_statusid in VariantOnTranscriptUnique view may contain
                        //       multiple IDs in a string. Therefore only the first char is used.
                        $nRowStatus = min($nRowStatus, intval(substr($zData[$sField], 0, 1)));
                    }
                }
            }

            // Determine row's CSS class name based on row status.
            $zData['class_name'] = '';
            if ($nRowStatus == STATUS_MARKED) {
                $zData['class_name'] = 'marked';
            } elseif ($nRowStatus < STATUS_MARKED) {
                $zData['class_name'] = 'del';
            }
~~~

Other than that, I don't see any problems merging as it is.

Reply:
Thanks! I've used your code (some edits to simplify and keep some of my comments) in 89f4029.